dbms/src/Storages/StorageMergeTree.cpp

```diff
@@ -221,14 +259,7 @@ void StorageMergeTree::alter(
     ASTPtr new_primary_key_ast = data.primary_key_ast;
     params.apply(new_columns, new_indices, new_order_by_ast, new_primary_key_ast);
 
-    auto parts = data.getDataParts({MergeTreeDataPartState::PreCommitted, MergeTreeDataPartState::Committed, MergeTreeDataPartState::Outdated});
-    auto columns_for_parts = new_columns.getAllPhysical();
-    std::vector<MergeTreeData::AlterDataPartTransactionPtr> transactions;
-    for (const MergeTreeData::DataPartPtr & part : parts)
-    {
-        if (auto transaction = data.alterDataPart(part, columns_for_parts, new_indices.indices, false))
-            transactions.push_back(std::move(transaction));
-    }
+    auto transactions = prepareAlterTransactions(new_columns, new_indices, 2 * getNumberOfPhysicalCPUCores());
```

Comment:
No idea, why we have 2 here.
Maybe introduce a setting (`max_alter_threads`, similar to `max_threads`)?
(This setting will make sense for default (system) profile.)

Reply:
hyperthreading? 2 threads by core

I copypasted from 
dbms/programs/copier/ClusterCopier.cpp
```
ThreadPool thread_pool(num_threads ? num_threads : 2 * getNumberOfPhysicalCPUCores());
```

Reply:
Should we introduce 1 threadpool for all ALTER operations? Not only for Modify column?

Reply:
Answered verbally.
(no need to have a single ThreadPool)

Reply:
done